pkg/datapath/linux/node.go

```diff
@@ -609,11 +609,45 @@ func (n *linuxNodeHandler) insertNeighbor(newNode *nodeTypes.Node, ifaceName str
 	}
 
 	nextHopStr := nextHopIPv4.String()
+	if existingNextHopStr, found := n.neighNextHopByNode[newNode.Identity()]; found {
+		if existingNextHopStr == nextHopStr {
+			// We already know about the nextHop of the given newNode. Can happen
+			// when insertNeighbor is called by NodeUpdate multiple times for
+			// the same node.
+			if !refresh {
+				// In the case of refresh, don't return early, as we want to
+				// update the related neigh entry even if the nextHop is the same
+				// (e.g. to detect the GW MAC addr change).
+				return
+			}
+		} else if n.neighNextHopRefCount.Delete(existingNextHopStr) {
+			// nextHop has changed and nobody else is using it, so remove the old one.
+			neigh, found := n.neighByNextHop[existingNextHopStr]
+			if found {
+				if err := netlink.NeighDel(neigh); err != nil {
+					log.WithFields(logrus.Fields{
+						logfields.IPAddr:       neigh.IP,
+						logfields.HardwareAddr: neigh.HardwareAddr,
+						logfields.LinkIndex:    neigh.LinkIndex,
+					}).WithError(err).Warn("Failed to remove neighbor entry")
+				}
+				delete(n.neighByNextHop, nextHopStr)
```

Comment:
Should we still delete it even if the `netlink` returned an error?

Reply:
Yeah, as otherwise nobody would remove it from the map.